js/gestionPresupuestoWeb.js

```diff
@@ -81,11 +121,217 @@ function mostrarGastosAgrupadosWeb(idElemento, agrup, periodo)
+        let plantillaFormularioEditarGasto = document.getElementById("formulario-template");
+        let formularioEditarGasto = plantillaFormularioEditarGasto.content.cloneNode("true");
+        
+        e.target.disabled = "disabled";
+        
+        let manejadorEditarGasto = new editarGastoHandle();
+        manejadorEditarGasto.botonCrearFormulario = e.target;
+        
+        manejadorEditarGasto.gastoActual = this.gastoActual;
+        let botonEditarGasto = formularioEditarGasto.querySelector("form");
+        botonEditarGasto.addEventListener("submit", manejadorEditarGasto);
+        
+        let manejadorCancelar = new manejadorCancelarNuevoGastoWeb();
+        
+        manejadorCancelar.botonCrearFormulario = e.target;
+        let botonCancelar = formularioEditarGasto.querySelector("button.cancelar");
+        botonCancelar.addEventListener("click", manejadorCancelar);
+       
+        formularioEditarGasto.getElementById("descripcion").value = this.gastoActual.descripcion;
+        formularioEditarGasto.getElementById("valor").value = this.gastoActual.valor;
+        let fecha = new Date(this.gastoActual.fecha);
+        formularioEditarGasto.getElementById("fecha").value = fecha.toISOString().substr(0,10);
+        
+        formularioEditarGasto.getElementById("etiquetas").value = this.gastoActual.etiquetas;
+        
+        e.target.parentNode.append(formularioEditarGasto);
+        
+        
+    }
+}
+function editarGastoHandle() {
+    this.handleEvent = function(e) {
+        
+        e.preventDefault();
+        if (e.target.elements.descripcion.value != this.gastoActual.descripcion || 
+            e.target.elements.descripcion.value != "") {
+                this.gastoActual.actualizarDescripcion(e.target.elements.descripcion.value);
+        }
+        
+        if (parseFloat(e.target.elements.valor.value) != this.gastoActual.valor || 
+            parseFloat(e.target.elements.valor.value) != "") {
+                this.gastoActual.actualizarValor(parseFloat(e.target.elements.valor.value));
+            }
+        
+        if (e.target.elements.fecha.value != this.gastoActual.fecha || 
+            e.target.elements.fecha.value != "") {
+                this.gastoActual.actualizarFecha(e.target.elements.fecha.value);
+            }
+                
+        let nuevasEtiq = e.target.elements.etiquetas.value;
+        nuevasEtiq = nuevasEtiq.split(",");
+        
+        if (e.target.elements.etiquetas.value != "") {
+            this.gastoActual.anyadirEtiquetas(nuevasEtiq);
+            this.gastoActual.anyadirEtiquetas(...nuevasEtiq);
+        }
+                
+        repintar();
+        e.target.remove();
+        
+    }
+}
+function BorrarHandle(){
+    this.handleEvent = function(e){
+
+        gestionPres.borrarGasto(this.gasto.id);
+
+        repintar();
+    }
+}
+
+function BorrarEtiquetasHandle(){
+    this.handleEvent = function(e){
+
+        this.etiqueta = this.etiqueta.split(',');
```

Comment:
No es necesario, ya que `this.etiqueta` es una única etiqueta. Se la puedes pasar directamente a `borrarEtiquetas`.